src/sicar.py

```diff
@@ -254,4 +254,8 @@ def download_all_from_list(self, base_folder: str = None, debug: bool = False):
         for state in self.__states:
             folder="{0}/{1}".format(base_folder,state)
             Path(folder).mkdir(parents=True, exist_ok=True)
-            self.download_state(state=state, folder=folder, debug=debug)
\ No newline at end of file
+            details=self.download_state(state=state, folder=folder, debug=debug)
+            if isinstance(details, dict):
+                # store log file with failed codes
+                with open(folder+'/failed_codes.txt', 'w') as f:
```

Comment:
```suggestion
Improve log using logging library

```

Reply:
I usually include a logging library when I need to implement a global strategy to write records to all code points, with exceptions that occur like stack tracing.

In this particular case, when I want to store the codes that supposedly fail in captcha, I like to keep it simple.

**But your repository, your code, your rules.**

Testing the outputs, I noticed that the stored municipal codes failed because the files have a problem in the source, SFB site, not the captcha driver. The Tesseract solves this case exquisitely.